server/src/util/foodEntryMiddleware.js

```diff
@@ -53,40 +53,176 @@ export const isExpired = (req, res, next) => {
+ * @param {object} res - response
+ * @param {object} next - call back function
+ */
+ export const isRack = async (req, res, next) => {
+  const { rackId } = req.body;
+  const query = await prisma.foodEntry.findUnique({
+    where: {
+      id: rackId,
+    },
+  });
+  if (query === NULL || query === undefined) {
+    return res
+      .status(StatusCodes.CONFLICT)
+      .json({ msg: "ERROR: rack does not exist" });
+  } else {
+    next();
+  }
+};
+
+/**
+ * checks if the category exists
+ * @param {object} req - request
+ * @param {object} res - response
+ * @param {object} next - call back function
+ */
+ export const isCategory = async (req, res, next) => {
+  const { categoryId } = req.body;
+  const query = await prisma.foodEntry.findUnique({
+    where: {
+      id: categoryId,
+    },
+  });
+  if (query === NULL || query === undefined) {
+    return res
+      .status(StatusCodes.CONFLICT)
+      .json({ msg: "ERROR: category does not exist" });
+  } else {
+    next();
+  }
+};
+
+/**
+ * checks if weight is positive
+ * @param {object} req - request
+ * @param {object} res - response
+ * @param {object} next - call back function
+ */
+ export const isPositiveWeight = (req, res, next) => {
+  const { weight } = req.body;
+  if (weight < 0) {
+    res
+      .status(StatusCodes.BAD_REQUEST)
+      .json({
+        msg: "ERROR: weight is less than 0",
+      });
+  } else {
+    next();
+  }
+};
+
+/**
+ * checks if rackId exists if onRack is false
+ * @param {object} req - request
+ * @param {object} res - response
+ * @param {object} next - call back function
+ */
+ export const isOnRackFalse = (req, res, next) => {
+  const { onRack, rackId } = req.body;
+  const query = await prisma.foodEntry.findUnique({
+    where: {
+      onRack,
+      id: rackId,
+    },
+  });
+  if (!onRack && !(query === NULL || query === undefined)) {
+    res
+      .status(StatusCodes.CONFLICT)
+      .json({
+        msg: "ERROR: rack given but on rack is false",
+      });
+  } else {
+    next();
+  }
+};
+
+/**
+ * checks if rackId exists if onRack is true
+ * @param {object} req - request
+ * @param {object} res - response
+ * @param {object} next - call back function
+ */
+ export const isOnRackTrue = (req, res, next) => {
+  const { onRack, rackId } = req.body;
+  const query = await prisma.foodEntry.findUnique({
+    where: {
+      onRack,
+      id: rackId,
+    },
+  });
+  if (onRack && !(query === NULL || query === undefined)) {
+    res
+      .status(StatusCodes.CONFLICT)
+      .json({
+        msg: "ERROR: rack location is not given",
+      });
+  } else {
+    next();
+  }
+};
+
+/**
+ * checks rack true --> warehouse true
+ * @param {object} req - request
+ * @param {object} res - response
+ * @param {object} next - call back function
+ */
+ export const isWarehouseTrue = (req, res, next) => {
+  const { inWarehouse, onRack } = req.body;
+  const query = await prisma.foodEntry.findUnique({
+    where: {
```

Comment:
you do not need the query here. Plus querying for a unique object using inWarehouse and onRack will throw an error since neither of these fields are unique. Although you do not need the query here, if you wanted one, you could use the findFirst instead of findUnique to query for the first entry given specific non-unique attributes